### PR TITLE
logging: implement invenio-sip2 logger

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds invenio-sip2 logger.
* Adds console logging configuration.
* Adds file system logging configuration.
* Uses new logger for logging.
* Adds a check when trying to start a new server.
* Adds permissions factory.

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -118,32 +116,32 @@ def _app_factory(config=None):
...
-                login_handler=remote_login_failed_handler,
-            ),
+            )
         )
     )
-    Babel(app)
-    InvenioDB(app)
-    InvenioAccess(app)
-    InvenioAccounts(app)
+    Babel(app_)
```
Comment (jma) [inline tests/conftest.py:122]: I do not understand why the app name has been changed.

**Review by jma — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

File: invenio_sip2/actions/actions.py [hunk 2]
```diff
@@ -384,8 +406,9 @@ def execute(self, message, client, **kwargs):
         patron_id = message.get_field_value('patron_id')
 
         hold = hold_handler(
-            client.remote_app, client.user_id, client.institution_id,
-            patron_id, item_id, language=patron_session.get('language')
+            client.remote_app, client.transaction_user_id,
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: you are right! It's done.

File: invenio_sip2/actions/actions.py [hunk 3]
```diff
@@ -419,8 +442,9 @@ def execute(self, message, client, **kwargs):
         patron_id = message.get_field_value('patron_id')
 
         renew = renew_handler(
-            client.remote_app, client.user_id, client.institution_id,
-            patron_id, item_id, language=patron_session.get('language')
+            client.remote_app, client.transaction_user_id,
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: It's done.

File: invenio_sip2/cli.py [hunk 4]
```diff
@@ -54,10 +56,16 @@ def selfcheck():
 def start_socket_server(name, host, port, remote):
     """Start sockets server with unique name."""
     try:
-        # TODO: check if server already exist and running
+        server = Server.find_server(server_name=name)
+        if server and server.is_running:
+            raise ServerAlreadyRunning(
+                'server [{name}] already running on {port}'.format(
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/cli.py]: Ok.

File: invenio_sip2/ext.py [hunk 5]
```diff
@@ -93,6 +120,20 @@ def init_config(self, app):
         load_fixed_field(app)
         load_variable_field(app)
 
+    def add_logger_handler(self, handler, level):
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/ext.py]: ok

File: invenio_sip2/config.py [hunk 6]
```diff
@@ -118,17 +118,38 @@
...
+
+"""Enable logging to the filesystem."""
+SIP2_LOGGING_FS_BACKUPCOUNT = 5
+"""Number of rotated log files to keep."""
+
+SIP2_LOGGING_FS_MAXBYTES = 100 * 1024 * 1024
+"""Maximum size of logging file. Default: 100MB."""
+
+SIP2_PERMISSIONS_FACTORY = default_permission_factory
+"""."""
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/config.py]: done

File: invenio_sip2/config.py [hunk 7]
```diff
@@ -118,17 +118,38 @@
...
+# DATASTORE
+# ====
 SIP2_DATASTORE_HANDLER = 'invenio_sip2.datastore:Sip2RedisDatastore'
 SIP2_DATASTORE_REDIS_PREFIX = 'sip2'
 SIP2_DATASTORE_REDIS_URL = 'redis://localhost:16379/0'
 
+# LOGGING
+# ====
+
+SIP2_LOGGING_CONSOLE = True
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/config.py:139]: done

File: tests/conftest.py (hunk 1 above)
Comment (lauren-d) [reply tests/conftest.py]: oops, it's a mistake

File: invenio_sip2/actions/actions.py [hunk 8]
```diff
@@ -295,13 +303,20 @@ def execute(self, message, client, **kwargs):
...
-        patron_id = message.get_field_value('patron_id')
 
-        checkin = checkin_handler(
-            client.remote_app, client.user_id, client.institution_id,
-            patron_session.get('patron_id'), item_id,
-            language=patron_session.get('language')
-        )
+        try:
+            checkin = checkin_handler(
+                client.remote_app, client.transaction_user_id,
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply invenio_sip2/actions/actions.py:310]: ok, i will do that in next pull request

**Pull request merged**